connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/a2a/client/model/result/A2AClientSendMessageResult.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.agenticai.a2a.client.model.result;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import io.camunda.connector.agenticai.model.AgenticAiRecord;
+import io.camunda.connector.agenticai.model.message.content.Content;
+import java.util.List;
+
+@AgenticAiRecord
```

Comment:
I'd suggest we also add the `@JsonDeserialize` config to all the models to force using the builder as done for most of the models (example: https://github.com/camunda/connectors/blob/main/connectors/agentic-ai/src/main/java/io/camunda/connector/agenticai/model/message/SystemMessage.java#L29). 

I raised an issue to add this to record-builder some time ago, but for the moment we need to manually create this setup. By forcing deserialization through the builder we have a couple of benefits without implementing it in the record constructor:

- ensures empty collections are initialized vs. being `null`
- ensures non-nullable fields are not null
- allows setting defaults if needed